### PR TITLE
Update docs for no google groups support in ID Token mode

### PR DESCRIPTION
**Pull request by grjones**
Bearer tokens are not supported in the Google provider for group validation, so updating the docs.

## Description

For the Google provider, attempting to send `Authorization: <ID Token>` instead of using cookie-based auth when allowed_groups is enabled will not work. The error you get is:

```
[AuthFailure] Invalid authorization via session: removing session Session{email:SERVICE_ACCOUNT@PROJECT.iam.gserviceaccount.com user:XXX PreferredUsername: token:true id_token:true expires:2022-09-26 18:55:53 +0000 UTC}
```

Apparently this used to work, but the feature was quickly removed because it was validating the token on every request. See https://github.com/oauth2-proxy/oauth2-proxy/issues/433.
 

## Motivation and Context

My team spent a lot of effort trying to get this to work thinking we were misconfiguring something. Adding this to the docs will save others those same efforts.

## How Has This Been Tested?

Just docs. The GitHub Issues history and first-hand experience shows this is the case.

## Checklist:

- [ ] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by grjones** [comment]
Hi @JoelSpeed, any reason we can't merge in this small doc change? Hoping to prevent some other folks from going down the path I did.

**Comment by JoelSpeed** [comment]
@grjones Apologies, I missed this. If you can update the PR then we can get this added in assuming the statement is still correct

**Comment by grjones** [comment]
Thanks @JoelSpeed. No more conflicts. 😄

File: docs/versioned_docs/version-7.5.x/configuration/auth.md
Comment (JoelSpeed) [inline docs/versioned_docs/version-7.5.x/configuration/auth.md:1]: If you want this to be updated in the future, as in, kept for future releases, you also need to update the no versioned version in `docs/docs/configuration/auth.md`, can you update?

Comment (kvanzuijlen) [reply docs/versioned_docs/version-7.5.x/configuration/auth.md:1]: @grjones can you please update `docs/docs/configuration/providers/google.md` to include this change as well?

**Pull request closed without merging**